viabel/optimization.py

```diff
@@ -238,7 +393,7 @@ def __init__(self, sgo, *, mcse_threshold=0.1, W_min=200, ESS_min=None,
             raise ValueError('"ESS_min" must be greater than zero')
 
     def optimize(self, n_iters, objective, init_param):
-        dim = init_param.size
+        dim = int(init_param.size/2)
```

Comment:
this doesn't seem like a good way to determine the dimension as it only works for MF Gaussian family